lib/types/index.ts

```diff
@@ -76,3 +76,18 @@ export type InitialConfig = {
   console?: ConsoleConfig;
   file?: FileConfig;
 };
+
+export type PluginConfig = {
+  silent: boolean,
+  level: LogLevel,
+  prettify?: boolean,
+  [property: string]: any
+}
+
+export type LoggerPlugin = {
+  [property: string]: any
+  debug: (...content: any[]) => void,
+  info: (...content: any[]) => void,
+  warning: (...content: any[]) => void,
```

Comment:
use `warn` instead of `warning`, better saying prefer using `LogLevel` enum to refer to this methods like:
` [LogLevel.debug]: (...content: any[]) => void`

Reply:
this methods should receive only 1 argument

Reply:
Are you sure that "this methods should receive only 1 argument"?